great_expectations_cloud/agent/actions/run_column_descriptive_metrics_action.py

```diff
@@ -25,12 +34,16 @@ class ColumnDescriptiveMetricsAction(AgentAction[RunColumnDescriptiveMetricsEven
     def __init__(
         self,
         context: CloudDataContext,
-        metric_repository: MetricRepository,
-        batch_inspector: BatchInspector,
+        metric_repository: MetricRepository | None = None,
+        batch_inspector: BatchInspector | None = None,
     ):
         super().__init__(context=context)
-        self._metric_repository = metric_repository
-        self._batch_inspector = batch_inspector
+        metric_retrievers: list[MetricRetriever] = [
```

Comment:
This is only used when defining `self._batch_inspector` so, to prevent creating it when you don't need it, you could delete this and change line 46 to:
```
self._batch_inspector = batch_inspector or BatchInspector(context, ColumnDescriptiveMetricsMetricRetriever(self._context))
```

Reply:
Good call, same thing with `CloudDataStore`. Fixed!